pkg/diagnostics/cluster/registry.go

```diff
@@ -338,8 +338,11 @@ func (d *ClusterRegistry) verifyRegistryImageStream(service *kapi.Service, r typ
 	}
 	r.Debug("DClu1018", fmt.Sprintf("Created test ImageStream: %[1]v", imgStream))
 	cacheHost := strings.SplitN(imgStream.Status.DockerImageRepository, "/", 2)[0]
-	serviceHost := fmt.Sprintf("%s:%d", service.Spec.ClusterIP, service.Spec.Ports[0].Port)
-	if cacheHost != serviceHost {
-		r.Error("DClu1019", nil, fmt.Sprintf(clRegISMismatch, registryName, serviceHost, cacheHost))
+	// the registry for imagestreams was previously recorded as an IP, which could change if the registry service were re-created.
+	// Now it is a cluster hostname, which should be unchanging even if re-created. Just ensure it is the right hostname.
+	serviceIpPort := fmt.Sprintf("%s:%d", service.Spec.ClusterIP, service.Spec.Ports[0].Port)
+	serviceHostPort := fmt.Sprintf("%s.%s.svc:%d", registryName, service.ObjectMeta.Namespace, service.Spec.Ports[0].Port)
+	if cacheHost != serviceIpPort && cacheHost != serviceHostPort {
+		r.Error("DClu1019", nil, fmt.Sprintf(clRegISMismatch, registryName, serviceIpPort, cacheHost))
```

Comment:
Should the message say something about serviceHostPort?

Reply:
I guess that would make sense, thanks.